### DISCUSSION TITLE
Deterministic self test

### DISCUSSION THREAD

**Discussion by kinmic** [ss3sim/ss3sim#420] [Q&A] [answered]
I'm currently trying to set up a deterministic model run for a new scenario. Based on the vignette I use the following:
```
df <- setup_scenarios_defaults(nscenarios = 2)
df[, "user_recdevs"] <- "matrix(0, nrow = 200, ncol = 20)"
df[, "co.par_name"] <- "SR_sigmaR"
df[, "co.par_int"] <- 0.001
df[, "ce.par_name"] <- "SR_sigmaR"
df[, "ce.par_int"] <- 0.001
df[, "ce.forecast_num"] <- 0
df[, "si.years.2"] <- "seq(60, 100, by = 2)"
df[, "si.sds_obs.2"] <- 0.001
df[, "ce.par_name"] <- "NatM_uniform_Fem_GP_1"
df[, "ce.par_int"] <- NA
df[, "ce.par_phase"] <- c(-1, 3)
df[, "scenarios"] <- c("D1-E100-F0-cod", "D1-E101-F0-M1-cod")
df[, "bias_adjust"] <- FALSE``

```
However is the changing of sigmaR not overwritten by the changing of natural mortality? The resulting df only has natural mortality for the ```"ce.par_name``` argument, so how is sigmaR changed in the simulation?

**Comment by kellijohnson-NOAA** [comment]
Thank you @kinmic for the question (which I have now moved to a Discussion - Question and Answer, rather than an Issue). The [Introduction vignette](https://ss3sim.github.io/ss3sim/articles/introduction.html) has an example for when you want to change a single parameter in your estimation method. If I am understanding correctly, you would like to change more than one parameter in each of your scenarios?
Below is the code as it stands in the vignette
```
df <- rbind(df, df)
df[, "ce.par_name"] <- "NatM_uniform_Fem_GP_1"
df[, "ce.par_int"] <- NA
# Set the phase for estimating natural mortality
df[, "ce.par_phase"] <- rep(c(-1, 3), each = 2)
# Name the scenarios
df[, "scenarios"] <- c(
  "D0-E0-F0-cod", "D1-E0-F0-cod",
  "D0-E1-F0-cod", "D1-E1-F0-cod"
)
```
But for your code, to change multiple parameters in the first scenario you will need to use single quotes around a vector if there are quotes in the vector. The vector that you supply for the other estimation method arguments must be as long as the vector of parameter names. Again, note that the example below is just to change your first scenario. If you wanted to do this for both scenarios just remove the indexing from the left side of the operator and it will supply the same information to both scenario 1 and scenario 2.
```
df[1, "ce.par_name"] <- 'c("SR_sigmaR", "NatM_uniform_Fem_GP_1")'
df[1, "ce.par_int"] <- "c(0.001, NA)"
```
Please let me know if you have any additional questions. The Discussion board is a great place to get help and the Issues board can be searched for known bugs in the code or desired feature requests.

**Reply by kinmic** [reply]
Hi Kelly,

Thanks for the quick reply. So just to check, in the vignette section [set up self test ](https://ss3sim.github.io/ss3sim/articles/introduction.html#setup-self-test) calls ``df[, "ce.par_name"] `` **twice**, first for sigmaR , then for natural mortality, which overwrite the the sigmaR parameter name in the df. So for a deterministic run, do I need to change the sigmaR parameter to 0.001 in the em in addition to the om, or just the om?


As per the vignette::
```
df <- setup_scenarios_defaults(nscenarios = 2)
df[, "user_recdevs"] <- "matrix(0, nrow = 200, ncol = 20)"
df[, "co.par_name"] <- "SR_sigmaR"
df[, "co.par_int"] <- 0.001
**df[, "ce.par_name"] <- "SR_sigmaR"**
df[, "ce.par_int"] <- 0.001
df[, "ce.forecast_num"] <- 0
df[, "si.years.2"] <- "seq(60, 100, by = 2)"
df[, "si.sds_obs.2"] <- 0.001
**df[, "ce.par_name"] <- "NatM_uniform_Fem_GP_1"**
df[, "ce.par_int"] <- NA
df[, "ce.par_phase"] <- c(-1, 3)
df[, "scenarios"] <- c("D1-E100-F0-cod", "D1-E101-F0-M1-cod")
df[, "bias_adjust"] <- FALSE
```

**Reply by kellijohnson-NOAA** [reply, marked as answer]
Ahh, I see. This is in fact a bug in the vignette. Thank you for bringing that to my attention. No wonder things were not working for you, again my apologies. The code should be
```
df <- setup_scenarios_defaults(nscenarios = 2)
df[, "user_recdevs"] <- "matrix(0, nrow = 200, ncol = 20)"
df[, "co.par_name"] <- "SR_sigmaR"
df[, "co.par_int"] <- 0.001
df[1, "ce.par_name"] <- "SR_sigmaR"
df[1, "ce.par_int"] <- 0.001
df[1, "ce.par_phase"] <- -1
df[, "ce.forecast_num"] <- 0
df[, "si.years.2"] <- "seq(60, 100, by = 2)"
df[, "si.sds_obs.2"] <- 0.001
df[2, "ce.par_name"] <- 'c("SR_sigmaR", "NatM_uniform_Fem_GP_1")'
df[2, "ce.par_int"] <- "c(0.001, NA)"
df[2, "ce.par_phase"] <- "c(-1, 3)"
df[, "scenarios"] <- c("D1-E100-F0-cod", "D1-E101-F0-M1-cod")
df[, "bias_adjust"] <- FALSE
```

The code was fixed with c54bd138

**Reply by kinmic** [reply]
No problem Kelly, thanks for the help!
